File: interview questions/runLengthEncoding.py
def runLengthEncoding(string):
    counter = 1
      result = ""

       for i in range(1, len(string)+1):
            prev = string[i-1]
            curr = string[i] if i < len(string) else None

            if (prev != curr or counter == 9):
                encoding = f'{counter}{prev}'
                result += encoding
                counter = 0

            counter += 1

        return result

File: interview questions/runLengthEncoding.js
function runLengthEncoding(string) {
  // have a counter variable and result string
  let counter = 1;
  let encodingResult = "";

  // loop through string
  for (let i = 1; i <= string.length; i++) {
    let prevChar = string[i - 1];
    let currentChar = string[i];

    // if different character or limit of 9 reached
    if (prevChar !== currentChar || counter === 9) {
      // generate encoding + reset counter
      let encoding = counter + prevChar;
      encodingResult += encoding;
      counter = 0;
    }

    counter++;
  }
  // return result
  return encodingResult;
}
